sled-agent/src/ledger.rs

```diff
@@ -58,15 +61,26 @@ pub struct Ledger<T> {
 }
 
 impl<T: Ledgerable> Ledger<T> {
+    /// Creates a ledger with a new initial value, ready to be written to
```

Comment:
I sorta needed to change these APIs for the RSS markers to work okay. Basically, if there was no file, I wanted to *know* about it, rather than getting a "default" value back. 

This PR removes the constraint that `Ledgerable` also implement `Default`, and updates these constructors to cope.